src/libraries/System.Private.CoreLib/src/System/SpanHelpers.T.cs

```diff
@@ -3,16 +3,18 @@
 // See the LICENSE file in the project root for more information.
 
 using System.Diagnostics;
-#if !NETSTANDARD2_0
+
 using Internal.Runtime.CompilerServices;
-#endif
 
 namespace System
 {
     internal static partial class SpanHelpers // .T
     {
         public static int IndexOf<T>(ref T searchSpace, int searchSpaceLength, ref T value, int valueLength) where T : IEquatable<T>
         {
+            // The optimized implementation should be used for these types
+            Debug.Assert(typeof(T) != typeof(byte) && typeof(T) != typeof(char));
```

Comment:
The optimized implementation is based on the size of T, so the assert should be
```suggestion
            Debug.Assert(Unsafe.SizeOf<T>() != sizeof(byte) && Unsafe.SizeOf<T>() != sizeof(char));
```
as `sbyte`, `short`, `ushort` will also take the optimized version.